pkgs/tools/misc/starship/default.nix

```diff
@@ -0,0 +1,26 @@
+{ stdenv, fetchFromGitHub, rustPlatform, openssl, pkgconfig, libiconv, darwin }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "starship";
+  version = "0.10.1";
+
+  src = fetchFromGitHub {
+    owner = "starship";
+    repo = "starship";
+    rev = "v${version}";
+    sha256 = "045lq4nsrdssmqbcj0551f2c5qd2rcvhs8gr4p4iniv7s89yz1xl";
+  };
+
+  buildInputs = [ openssl ] ++ stdenv.lib.optional stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
```

Comment:
this will return `[ openssl [ libiconv darwin.apple_sdk.frameworks.Security ]]` as `optional makes it's input a singleton list.
I think you instead want to use `stdenv.lib.optionals` here

Reply:
Great catch!

Reply:
Thanks!

Reply:
Interesting - the fact that this worked _anyway_ implies that `buildInputs` recurses into lists when evaluating, which is something I never knew it did.

Reply:
#60442